### PR TITLE
Pagination for apps and db operations pages

### LINKED ISSUE #49: display pagination for app/ and db/ operations lists
The API will return pagination info for collection resources (see https://github.com/aptible/api.aptible.com/pull/146). Diesel should use these to display pagination links, starting with the `apps/:id/operations and` and `databases/:id/operations` history pages.

### PR DESCRIPTION
**Pull request by bantic**
@mixonic for your review

  * Override findQuery for operation adapter to allow pagination
  * add pagination in {{aptible-pagination}} component
  * Paginated route and controller mixins for shared behaviors
  * update 'page' query param in url
  * dry tests with shared test helper functions

fixes #49 refs #51

cc @fancyremarker

This uses the current_page, total_count and per_page meta data
provided by the API as of https://github.com/aptible/api.aptible.com/pull/146

### CODE REVIEW

File: app/apps/show/operations/controller.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Ember from 'ember';
+import Paginated from 'diesel/mixins/controllers/paginated';
```
Comment (mixonic) [inline app/apps/show/operations/controller.js]: relative imports are generally preferred, as they avoid sprinkling the app name around and make the code more portable. At least, this has been the idiom so far.

File: app/databases/show/operations/route.js [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import Ember from 'ember';
+import Paginated from "diesel/mixins/routes/paginated";
 
-export default Ember.Route.extend({
-  model: function(){
+export default Ember.Route.extend(Paginated, {
+  model: function(params, transition){
     var db = this.modelFor('databases.show');
+    var page = parseInt(transition.queryParams.page || 1, 10);
```
Comment (mixonic) [inline app/databases/show/operations/route.js]: I understood that query params are always coerced to the same type as their original value (in this case `1` in the paginated mixin).

Also I believe the query param is on `params`, and accessing the transition is unnecessary. I can confirm.

File: app/mixins/routes/paginated.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    goToPage: function(page){
+      var controller = this.controller;
+      var store = this.store;
+      var route = this;
+      var query = { page: page };
+
+      var paginatedResourceOwner = this.controller.get('paginatedResourceOwner');
+      query[ paginatedResourceOwner.constructor.typeKey ] = paginatedResourceOwner;
+
+      store.find('operation', query).then(function(operations){
```
Comment (mixonic) [inline app/mixins/routes/paginated.js]: If this is a generic mixin, it should likely not reference `'operation'`

File: app/apps/show/operations/controller.js (hunk 1 above)
Comment (bantic) [inline app/apps/show/operations/controller.js]: Agreed, although in some cases I find the absolute paths like these easier to use intra codebase. The relative paths are non-copy-pastable if the files are in different folder paths. I.e., to use Paginated in the apps/index route you'd `import Paginated from "../../mixins/routes/paginated"` but for the apps/operations/index it is `import Paginated from "../../../mixins/routes/paginated"`, while `import Paginated from "diesel/mixins/routes/paginated"` works for both. The volatility of import namespace (diesel) feels lower to me than volatility of folder nesting. But I am not religious about this.

File: app/mixins/routes/paginated.js (hunk 3 above)
Comment (mixonic) [inline app/mixins/routes/paginated.js]: I might expect that the query to load a new page is a hook called on the route. So a route must include the mixin, and define a hook for fetching new pages. That way all the `store` related work stays on the route and this mixin is only responsible for managing convenience actions (next/prev page) and passing params to hooks on the route.

File: app/apps/show/operations/controller.js (hunk 1 above)
Comment (mixonic) [inline app/apps/show/operations/controller.js]: Aye. I would like to ponder a heuristic for when to use absolute vs relative.

File: app/databases/show/operations/route.js (hunk 2 above)
Comment (bantic) [inline app/databases/show/operations/route.js]: that's correct — the queryParam is coerced into params as its property name. In this case `params.currentPage`. I'll push a commit in a second.

**Comment by bantic** [comment]
closed in favor of #60

**Pull request closed without merging**